localstack/services/iam/resource_providers/aws_iam_accesskey.py

```diff
@@ -0,0 +1,128 @@
+
+
+class IAMAccessKeyProvider(ResourceProvider[IAMAccessKeyProperties]):
+
+    TYPE = "AWS::IAM::AccessKey"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[IAMAccessKeyProperties],
+    ) -> ProgressEvent[IAMAccessKeyProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Id
+
+        Required properties:
+          - UserName
+
+        Create-only properties:
+          - /properties/UserName
+          - /properties/Serial
+
+        Read-only properties:
+          - /properties/SecretAccessKey
+          - /properties/Id
+
+        """
+        # TODO: what alues can model['Serial'] take on initial create?
+        model = request.desired_state
+        iam_client = request.aws_client_factory.iam
+
+        access_key = iam_client.create_access_key(UserName=model["UserName"])
+        model["SecretAccessKey"] = access_key["AccessKey"]["SecretAccessKey"]
+        model["Id"] = access_key["AccessKey"]["AccessKeyId"]
+
+        if model.get("Status") == "Inactive":
+            # can be "Active" or "Inactive"
+            # by default the created access key has Status "Active", but if user set Inactive this needs to be adjusted
+            iam_client.update_access_key(
+                AccessKeyId=model["Id"], UserName=model["UserName"], Status=model["Status"]
+            )
+
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model=model)
+
+    def read(
+        self,
+        request: ResourceRequest[IAMAccessKeyProperties],
+    ) -> ProgressEvent[IAMAccessKeyProperties]:
+        """
+        Fetch resource information
+        """
+        raise NotImplementedError
+
+    def delete(
+        self,
+        request: ResourceRequest[IAMAccessKeyProperties],
+    ) -> ProgressEvent[IAMAccessKeyProperties]:
+        """
+        Delete a resource
+        """
+        iam_client = request.aws_client_factory.iam
+        model = request.previous_state
+        iam_client.delete_access_key(AccessKeyId=model["Id"], UserName=model["UserName"])
+        return ProgressEvent(status=OperationStatus.SUCCESS, resource_model={})
+
+    def update(
+        self,
+        request: ResourceRequest[IAMAccessKeyProperties],
+    ) -> ProgressEvent[IAMAccessKeyProperties]:
+        """
+        Update a resource
+        """
+        iam_client = request.aws_client_factory.iam
+
+        # FIXME: replacement should be handled in engine before here
+        user_name_changed = request.desired_state["UserName"] != request.previous_state["UserName"]
+        serial_changed = request.desired_state["Serial"] != request.previous_state["Serial"]
+        if user_name_changed or serial_changed:
+            # recreate the key
+            self.delete(request)
+            create_event = self.create(request)
+            return create_event
```

Comment:
Will basically be what the engine should then do, but for now we have to do it in the models. I think that is somewhat ok for now though since it *should* also not cause any issues when we test the provider against AWS since the if branch should never be true on AWS.